addons/mail/static/src/core/common/message_actions.js

```diff
@@ -68,17 +68,46 @@ registerMessageAction("reaction", {
     sequence: 10,
 });
 registerMessageAction("reply-to", {
-    condition: (component) => component.props.message.canReplyTo(component.props.thread),
+    condition: (component) => {
+        const { message, thread } = component.props;
+        return (
+            message.canReplyTo(thread) ||
+            (!["discuss.channel", "mail.box"].includes(thread.model) &&
+                message.isNote &&
+                !message.isSelfAuthored)
+        );
+    },
     icon: "fa fa-reply",
     iconLarge: "fa fa-lg fa-reply",
     name: _t("Reply"),
     onSelected: (component) => {
         const message = toRaw(component.props.message);
         const thread = toRaw(component.props.thread);
-        if (message.eq(thread.composer.replyToMessage)) {
-            thread.composer.replyToMessage = undefined;
+        const composer = thread.composer;
+        let currentText = composer.text || "";
+        const mentionText = `@${message.authorName} `;
+        if (message.eq(composer.replyToMessage)) {
+            composer.replyToMessage = undefined;
         } else {
-            thread.composer.replyToMessage = message;
+            if (["discuss.channel", "mail.box"].includes(thread.model)) {
+                composer.replyToMessage = message;
+            }
+            if (thread.model !== "discuss.channel") {
+                if (!message.isSelfAuthored && message.model !== "discuss.channel") {
+                    if (!composer.mentionedPartners.includes(message.author)) {
+                        composer.mentionedPartners.push(message.author);
+                    }
```

Comment:
`composer.mentionedPartners` looks like an array... But it's actually a record list.
We have nice features in a record list, such as:
```js
composer.mentionedPartners.add(message.author);
```
Yes, this `.add()` method behaves as `.push()` if record is not in record list else do nothing.

There's a `.delete()` that does the same thing.

Also FYI, records have a `.in()` method that works with collection of records such as record list. This can make the LOC more readable for english readers we all are. Instead of `composer.mentionedPartners.includes(message.author)`, you can write as `message.author.in(composer.mentionedPartners)`.